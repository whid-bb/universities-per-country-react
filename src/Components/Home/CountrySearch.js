/* eslint-disable react/jsx-props-no-spreading */
import { useDispatch } from 'react-redux';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Autocomplete from '@mui/material/Autocomplete';
import { useNavigate } from 'react-router-dom';
import countries from './countryB';
import { searchCountries } from '../../redux/countries/Countries';
import fetchCountries, { fetchCountriesDetails } from '../../redux/countries/CountriesService';

function CountrySearch() {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleClick = (val) => {
    if (val === null) {
      dispatch(fetchCountries());
      navigate('/');
    } else {
      dispatch(searchCountries(val));
      dispatch(fetchCountriesDetails(val.name));
      navigate(`/country-details/${val.name}`);
    }
  };

  return (
    <Autocomplete
      id="country-select-demo"
      options={countries}
      sx={{ width: '100%' }}
      onChange={(e, value) => handleClick(value)}
      autoHighlight
      getOptionLabel={(option) => option.name}
      renderOption={(props, option) => (
        <Box
          component="li"
          sx={{ '& > img': { mr: 2, flexShrink: 0 } }}
          {...props}
        >
          <img
            loading="lazy"
            width="20"
            src={`https://flagcdn.com/w20/${option.abbr.toLowerCase()}.png`}
            srcSet={`https://flagcdn.com/w40/${option.abbr.toLowerCase()}.png 2x`}
            alt=""
          />
          {option.name}
          {' '}
          (
          {option.abbr}
          ) +
          {option.code}
        </Box>
      )}
      renderInput={(params) => (
        <TextField
          {...params}
          size="small"
          label="Choose a country"
          inputProps={{
            ...params.inputProps,
            autoComplete: 'new-password',
          }}
        />
      )}
    />
  );
}

export default CountrySearch;
