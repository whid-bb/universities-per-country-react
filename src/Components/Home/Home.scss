.top-header {
  min-height: 15rem;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-between;
  padding: 1rem;
  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    background-image: url('../../img/universityBg.jpg');
    background-size: cover;
    background-repeat: no-repeat;
    background-position: 50% 100%;
    width: 100%;
    height: 100%;
    opacity: 0.2;
    z-index: -1;
  }

  .search-container {
    width: 100%;
    background-color: #fff;
    border-radius: 5px;
    padding: 10px;
  }
}

.top-bar {
  padding-inline: 20px;
  display: flex;
}

.home-link {
  margin-right: auto;
}
