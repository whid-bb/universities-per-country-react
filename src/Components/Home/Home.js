import { Outlet, Link, useMatch } from 'react-router-dom';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import KeyboardVoiceIcon from '@mui/icons-material/KeyboardVoice';
import HomeIcon from '@mui/icons-material/Home';
import SettingsIcon from '@mui/icons-material/Settings';
import List from '../List/List';
import CountrySearch from './CountrySearch';
import fetchCountries from '../../redux/countries/CountriesService';
import './Home.scss';

function Home() {
  const dispatch = useDispatch();

  const getHome = () => {
    dispatch(fetchCountries());
  };

  const match = useMatch('/');

  useEffect(() => {
    if (match !== null) dispatch(fetchCountries());
  }, []);
  return (
    <div>
      <div className="top-bar">
        <Link className="home-link" onClick={() => getHome()} to="/">
          <HomeIcon />
        </Link>
        <div>

          <KeyboardVoiceIcon />
          <SettingsIcon />
        </div>
      </div>
      <header className="top-header">
        <h2>Universities per country</h2>
        <div className="search-container">
          <CountrySearch />
        </div>
      </header>
      <List />
      <Outlet />
    </div>
  );
}

export default Home;
