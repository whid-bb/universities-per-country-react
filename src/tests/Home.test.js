import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../Components/Home/Home';

describe('Home test', () => {
  it('Home render correctly', () => {
    const HomePage = renderer.create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    ).toJSON();
    expect(HomePage).toMatchSnapshot();
  });

  test('Render Home component', () => {
    render(
      <Provider store={store}>
        <Router>

          <Home />
        </Router>
      </Provider>,
    );
  });
});
