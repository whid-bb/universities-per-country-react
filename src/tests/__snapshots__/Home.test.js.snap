// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test Home render correctly 1`] = `
<div>
  <div
    className="top-bar"
  >
    <a
      className="home-link"
      href="/"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="HomeIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </a>
    <div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardVoiceIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SettingsIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
        />
      </svg>
    </div>
  </div>
  <header
    className="top-header"
  >
    <h2>
      Universities per country
    </h2>
    <div
      className="search-container"
    >
      <div
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-dd2h8b-MuiAutocomplete-root"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="country-select-demo"
            id="country-select-demo-label"
          >
            Choose a country
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="new-password"
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="country-select-demo"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              readOnly={false}
              required={false}
              role="combobox"
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Choose a country
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div>
    <div
      className="country-list-container"
    />
  </div>
</div>
`;
